Tests/RevenueCatUITests/CustomerCenter/CustomerCenterViewModelTests.swift

```diff
@@ -128,6 +128,56 @@ class CustomerCenterViewModelTests: TestCase {
         }
     }
 
+    func testAppIsLatestVersion() {
+        let testCases = [
+            (currentVersion: "1.0.0", latestVersion: "2.0.0", expectedAppIsLatestVersion: false),
+            (currentVersion: "2.0.0", latestVersion: "2.0.0", expectedAppIsLatestVersion: true),
+            (currentVersion: "3.0.0", latestVersion: "2.0.0", expectedAppIsLatestVersion: true),
+            (currentVersion: "1.0.0", latestVersion: "1.1.0", expectedAppIsLatestVersion: false),
+            (currentVersion: "1.1.0", latestVersion: "1.1.0", expectedAppIsLatestVersion: true),
+            (currentVersion: "1.1.0", latestVersion: "1.0.0", expectedAppIsLatestVersion: true),
+            (currentVersion: "1.0.0", latestVersion: "1.0.1", expectedAppIsLatestVersion: false),
+            (currentVersion: "1.0.1", latestVersion: "1.0.1", expectedAppIsLatestVersion: true),
+            (currentVersion: "1.0.1", latestVersion: "1.0.0", expectedAppIsLatestVersion: true),
+            // The CFBundleVersion docs state:
+            // > You can include more integers but the system ignores them.
+            // https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion
+            // So we should do the same.
+            (currentVersion: "2.0.0.2.3.4", latestVersion: "2.0.0.3.4.5", expectedAppIsLatestVersion: true),
+            (currentVersion: "1.0.0.2.3.4", latestVersion: "2.0.0.3.4.5", expectedAppIsLatestVersion: false),
+            (currentVersion: "1.2", latestVersion: "2", expectedAppIsLatestVersion: false),
+            (currentVersion: "1.2", latestVersion: "1", expectedAppIsLatestVersion: true),
+            (currentVersion: "2", latestVersion: "1", expectedAppIsLatestVersion: true),
+            (currentVersion: "0", latestVersion: "1", expectedAppIsLatestVersion: false),
+            (currentVersion: "10.2", latestVersion: "2", expectedAppIsLatestVersion: true),
+            // We default to true if we fail to parse any of the two versions.
+            (currentVersion: "not-a-number", latestVersion: "not-a-number-either", expectedAppIsLatestVersion: true),
+            (currentVersion: "not-a-number", latestVersion: "1.2.3", expectedAppIsLatestVersion: true),
```

Comment:
how about `not.a.number`?

(it'd pass, but I mean just in case we accidentally break it in the future so that we look for words split by `.`)

Reply:
Good call, added! Also added checks for empty strings. https://github.com/RevenueCat/purchases-ios/pull/4169/commits/14732226881de97bc2a7e108e5d5666cb2d7fa3f